diff --git a/CMakeLists.txt b/CMakeLists.txt
index 76d0d06..f35261e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -24,6 +24,6 @@ else()
     MESSAGE(STATUS "SOQT_FOUND: " false)
 endif()
 
-MESSAGE(STATUS "SOQT_INCLUDE_DIR: " ${SoQt_INCLUDE_DIRS})
-MESSAGE(STATUS "SOQT_LIB_DIR: " ${SoQt_LIBRARY_DIRS})
+MESSAGE(STATUS "SOQT_INCLUDE_DIR: $ENV{PREFIX}/include")
+MESSAGE(STATUS "SOQT_LIB_DIR: $ENV{PREFIX}/lib")
 MESSAGE(STATUS "SOQT_VERSION: " ${SoQt_VERSION})
