diff --git a/interfaces/CMakeLists.txt b/interfaces/CMakeLists.txt
index 60c62a9..acdae2b 100644
--- a/interfaces/CMakeLists.txt
+++ b/interfaces/CMakeLists.txt
@@ -32,6 +32,7 @@ target_include_directories(coin
     )
 
 target_link_libraries(coin PUBLIC Coin::Coin)
+install(TARGETS coin DESTINATION ${python_libs}/pivy)
 
 
 if (SoQt_FOUND)
@@ -63,4 +64,5 @@ if (SoQt_FOUND)
         )
 
     target_link_libraries(soqt PUBLIC SoQt::SoQt)
+    install(TARGETS soqt DESTINATION ${python_libs}/pivy/gui)
 endif()
